Feature/av2 987/support multi day events

sasangachathumal opened the pull request:
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-987

**Additional Comments:**
Check Jira story to see changes.

**Checklist:**

- [x] Unit test completed?: (Y/N)
- [x] Docs folder up to date?: (Y/N)
- [x] Add if anything missed: (Y/N)

All checklists are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

sasangachathumal committed:
[AV2-987] added new field to event object.
created new mentods to dupicate event to each date.

sasangachathumal committed:
[AV2-987]
create method in util service to get date difference.
update event object in event list service.
update timeDisplayed method to show new time like design.
update event grouping in event list component.

sasangachathumal committed:
[AV2-987]
update event dates in details page for multi day event.
update item highlighting logic for multi day event.
done changes to show event day count with title.
update max event render count to 10.
added new data fields to event object.
update event sorting method.
updated list item component to show day count.

sasangachathumal committed:
[AV2-987] Added more unit test

sasangachathumal committed:
[AV2-987] added more unit test

sasangachathumal committed:
[AV2-987]more unit test.

sasangachathumal committed:
[AV2-987] more unit test

sasangachathumal committed:
[AV2-987]
add changes to fix sonar cloud issues.
added more unit test.

shawnm0705 commented on src/app/event-list/event-list.service.ts:
```diff
@@ -7,6 +7,8 @@ import { BrowserStorageService } from '@services/storage.service';
 import { NotificationService } from '@shared/notification/notification.service';
 import { EventDetailComponent } from '@app/event-detail/event-detail.component';
 
+import * as moment from 'moment';
```
we'd better import 3rd party package in `utils.service` and create a function there, so that we can reuse it everywhere else

sasangachathumal committed:
[AV2-987]
fixed date issue.
update unit test for new changes.
update the code of getting date difference.

shawnm0705 reviewed (changes requested):
Please check my comments and fix the code smell

sasangachathumal committed:
[AV2-987]
Fixed issue found in code review.
Fixed code smells found in sonar cloud.

sasangachathumal committed:
[AV2-987] fixing code smells
